File: app/src/main/java/com/example/architectureapplication/di/AppModule.java
package com.example.architectureapplication.di;

import android.app.Application;

import com.example.architectureapplication.NoteDao;
import com.example.architectureapplication.NoteDatabase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @Singleton
    @Provides
    public NoteDatabase provideDb(Application application){
        return NoteDatabase.getInstance(application);
    }

    @Singleton
    @Provides
    public NoteDao provideNoteDao(NoteDatabase db){
        return db.noteDao();
    }

}

File: app/src/main/java/com/example/architectureapplication/kt/AppModule.kt
package com.example.architectureapplication.kt

import android.app.Application
import com.example.architectureapplication.NoteDatabase
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Singleton
    @Provides
    fun provideDb(app: Application): NoteDatabase
            = NoteDatabase.getInstance(app)

    @Singleton
    @Provides
    fun provideNoteDao(db: NoteDatabase) = db.noteDao()
}
